v0.17.0-rc4: text objects of type "mrkdwn" must not include the 'emoji' field.

### What happened
The following code works fine on v0.16.0:

```
	_, _, err := c.PostMessage(chID, slack.MsgOptionBlocks(
		slack.NewSectionBlock(
			slack.NewTextBlockObject(slack.MarkdownType, "*this is a test message*", false, false),
			[]*slack.TextBlockObject{
				slack.NewTextBlockObject(slack.MarkdownType, "this is a test message line", false, false),
			},
			nil,
		),
	))
```

However, on v0.17.0-rc4 PostMessage fails with an error "invalid_blocks". 

Tracing through, slack rejects the message due to the presence of the emoji parameter, introduced in https://github.com/slack-go/slack/commit/8f006d10e5201175238367595da4b06c77c22036.


### Expected behavior
PostMessage should still work.

### Steps to reproduce
See test code above.

#### reproducible code
See test code above.

#### manifest.yaml

### Versions
- Go: 1.24.2
- slack-go/slack: v0.17.0-rc4

More details: the error's ResponseMetadata has the following Messages:

0 = {string} "[ERROR] failed to match all allowed schemas [json-pointer:/blocks/0/text]"
1 = {string} "[ERROR] invalid additional property: emoji [json-pointer:/blocks/0/text]"
2 = {string} "[ERROR] failed to match all allowed schemas [json-pointer:/blocks/0/fields/0]"
3 = {string} "[ERROR] invalid additional property: emoji [json-pointer:/blocks/0/fields/0]"

--- a/attachments_test.go
+++ b/attachments_test.go
@@ -19,7 +19,6 @@ func TestAttachment_UnmarshalMarshalJSON_WithBlocks(t *testing.T) {
         "text": {
           "type": "mrkdwn",
           "text": "Pick something:",
-          "emoji": null,
           "verbatim": true
         },
         "accessory": {

--- a/block_object.go
+++ b/block_object.go
@@ -122,7 +122,7 @@ func unmarshalBlockObject(r json.RawMessage, object blockObject) (blockObject, e
 type TextBlockObject struct {
 	Type     string `json:"type"`
 	Text     string `json:"text"`
-	Emoji    *bool  `json:"emoji"`
+	Emoji    *bool  `json:"emoji,omitempty"`
 	Verbatim bool   `json:"verbatim,omitempty"`
 }
 
@@ -142,9 +142,8 @@ func (s TextBlockObject) Validate() error {
 		return errors.New("type must be either of plain_text or mrkdwn")
 	}
 
-	// https://github.com/slack-go/slack/issues/881
-	if s.Type == "mrkdwn" && s.Emoji != nil && *s.Emoji {
-		return errors.New("emoji cannot be true in mrkdown")
+	if s.Type == "mrkdwn" && s.Emoji != nil {
+		return errors.New("emoji cannot be set for mrkdwn type")
 	}
 
 	// https://api.slack.com/reference/block-kit/composition-objects#text__fields
@@ -161,11 +160,28 @@ func (s TextBlockObject) Validate() error {
 }
 
 // NewTextBlockObject returns an instance of a new Text Block Object
+//
+// If you want to create a mrkdwn object, you should set the emoji parameter to false. The
+// reason is that Slack doesn't accept emoji in mrkdwn.
 func NewTextBlockObject(elementType, text string, emoji bool, verbatim bool) *TextBlockObject {
+	// If we're trying to build a mrkdwn object, we can't send emoji at all. I think the
+	// right approach here is to be a bit clever, and not break the function interface.
+	//
+	// So, here's the plan:
+	// 1. If the type is mrkdwn, set emoji to nil, regardless of what the user passed in
+	// 2. Else, set emoji to the value passed in
+	var emojiPtr *bool
+
+	if elementType == "mrkdwn" {
+		emojiPtr = nil
+	} else {
+		emojiPtr = &emoji
+	}
+
 	return &TextBlockObject{
 		Type:     elementType,
 		Text:     text,
-		Emoji:    &emoji,
+		Emoji:    emojiPtr,
 		Verbatim: verbatim,
 	}
 }

--- a/block_object_test.go
+++ b/block_object_test.go
@@ -117,15 +117,6 @@ func TestValidateTextBlockObject(t *testing.T) {
 			input: TextBlockObject{
 				Type:     "mrkdwn",
 				Text:     "testText",
-				Emoji:    emojiFalse,
-				Verbatim: false,
-			},
-			expected: nil,
-		},
-		{
-			input: TextBlockObject{
-				Type:     "mrkdwn",
-				Text:     "testText",
 				Emoji:    nil,
 				Verbatim: false,
 			},
@@ -147,13 +138,22 @@ func TestValidateTextBlockObject(t *testing.T) {
 				Emoji:    emojiTrue,
 				Verbatim: false,
 			},
-			expected: errors.New("emoji cannot be true in mrkdown"),
+			expected: errors.New("emoji cannot be set for mrkdwn type"),
+		},
+		{
+			input: TextBlockObject{
+				Type:     "mrkdwn",
+				Text:     "testText",
+				Emoji:    emojiFalse,
+				Verbatim: false,
+			},
+			expected: errors.New("emoji cannot be set for mrkdwn type"),
 		},
 		{
 			input: TextBlockObject{
 				Type:     "mrkdwn",
 				Text:     "",
-				Emoji:    emojiFalse,
+				Emoji:    nil,
 				Verbatim: false,
 			},
 			expected: errors.New("text must have a minimum length of 1"),
@@ -162,7 +162,7 @@ func TestValidateTextBlockObject(t *testing.T) {
 			input: TextBlockObject{
 				Type:     "mrkdwn",
 				Text:     strings.Repeat("a", 3001),
-				Emoji:    emojiFalse,
+				Emoji:    nil,
 				Verbatim: false,
 			},
 			expected: errors.New("text cannot be longer than 3000 characters"),
@@ -171,7 +171,7 @@ func TestValidateTextBlockObject(t *testing.T) {
 
 	for _, test := range tests {
 		err := test.input.Validate()
-		assert.Equal(t, err, test.expected)
+		assert.Equal(t, test.expected, err)
 	}
 }
 

--- a/block_section_test.go
+++ b/block_section_test.go
@@ -30,7 +30,7 @@ func TestNewBlockSectionContainsAddedTextBlockAndAccessory(t *testing.T) {
 	textBlockInSection := sectionBlock.Text
 	assert.Equal(t, textBlockInSection.Text, textBlockObject.Text)
 	assert.Equal(t, textBlockInSection.Type, textBlockObject.Type)
-	assert.True(t, *textBlockInSection.Emoji)
+	assert.Nil(t, textBlockInSection.Emoji)
 	assert.False(t, textBlockInSection.Verbatim)
 	assert.Equal(t, sectionBlock.Accessory.ImageElement, conflictImage)
 }

--- a/examples/blocks/README.md
+++ b/examples/blocks/README.md
@@ -9,7 +9,7 @@ The examples below should cover implementing most supported block elements.
 For additional information on Blocks, see the [Block Kit website](https://api.slack.com/block-kit).
 
 ### Using examples with the Block Kit Builder website
-When generating examples, they will be printed to the screen as a complete message that is meant to be sent back to slack as a direct response, or throuogh the ResponseURL provided.  To test your examples in the Block Kit Builder, you must take the contents of the `blocks` property and paste the results into the builder. 
+When generating examples, they will be printed to the screen as a complete message that is meant to be sent back to slack as a direct response, or throuogh the ResponseURL provided.  To test your examples in the Block Kit Builder, you must take the contents of the `blocks` property and paste the results into the builder.
 
 For example, when printing a simple header, the output will be
 

--- a/interactions_test.go
+++ b/interactions_test.go
@@ -61,8 +61,7 @@ const (
 				"type": "section",
 				"text": {
 				  "text": "*Sally* has requested you set the deadline for the Nano launch project",
-				  "type": "mrkdwn",
-				  "emoji": false
+				  "type": "mrkdwn"
 				},
 				"accessory": {
 				  "type": "datepicker",

